homeassistant/components/knx/storage/entity_store_schema.py

```diff
@@ -0,0 +1,81 @@
+"""KNX entity store schema."""
+
+import voluptuous as vol
+
+from homeassistant.const import Platform
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.entity import ENTITY_CATEGORIES_SCHEMA
+
+from ..const import SUPPORTED_PLATFORMS_UI
+from ..validation import sync_state_validator
+from .knx_selector import GASelector
+
+BASE_ENTITY_SCHEMA = vol.All(
+    {
+        vol.Optional("name", default=None): vol.Maybe(str),
+        vol.Optional("device_info", default=None): vol.Maybe(str),
+        vol.Optional("entity_category", default=None): vol.Any(
+            ENTITY_CATEGORIES_SCHEMA, vol.SetTo(None)
+        ),
+    },
+    vol.Any(
+        vol.Schema(
+            {
+                vol.Required("name"): str,
+            },
+            extra=vol.ALLOW_EXTRA,
+        ),
+        vol.Schema(
+            {
+                vol.Required("device_info"): str,
+            },
+            extra=vol.ALLOW_EXTRA,
+        ),
+        msg="One of `Device` or `Name` is required",
```

Comment:
the fields in there are `device_info` and `name` right? Is there a reason we call them `Device` and `Name` here? The backticks will make them look like they are code, but apparently they are not

Reply:
This refers to the 2 selectors in the frontend. One is a custom device selector (like an area selector where you can select a knx device, or create a new), the other is the entity name. The name selector is prefixed with the device name (in UI, not in schema). If there is no device selected, a user has to enter some entity name, if there is a device, it can be `None` (`_attr_has_entity_name`).
<img width="741" alt="Bildschirmfoto 2024-07-18 um 22 47 00" src="https://github.com/user-attachments/assets/00044b35-ff8a-45eb-8313-75d5ee18dfc6">

Reply:
Hmm, it's nasty because they can be translated

Reply:
The custom panel doesn't have translation in place yet. I guess this can be worked on when translation was implemented... for other things we use voluptuous error messages which are english too.